### PR TITLE
NGRAPH-1742 SoftmaxActivation 

### PR DESCRIPTION
**Pull request by mbrookhart**
SoftmaxActivation op, plus a test with values based on the base-CPU kernel. For one shape and option combination, MXNet's backward op is wrong so I disabled the test. This op is deprecated and untested, and all models we target avoid this particular combination.

### Essentials ###
Please feel free to remove inapplicable items for your PR.
- [ ] The PR title starts with [MXNET-$JIRA_ID], where $JIRA_ID refers to the relevant [JIRA issue](https://issues.apache.org/jira/projects/MXNET/issues) created (except PRs with tiny changes)
- [ ] Changes are complete (i.e. I finished coding on this PR)
- [ ] All changes have test coverage:
- Unit tests are added for small changes to verify correctness (e.g. adding a new operator)
- Nightly tests are added for complicated/long-running ones (e.g. changing distributed kvstore)
- Build tests will be added for build configuration changes (e.g. adding a new build option with NCCL)
- [ ] Code is well-documented: 
- For user-facing API changes, API doc string has been updated. 
- For new C++ functions in header files, their functionalities and arguments are documented. 
- For new examples, README.md is added to explain the what the example does, the source of the dataset, expected performance on test set and reference to the original paper if applicable
- Check the API doc at http://mxnet-ci-doc.s3-accelerate.dualstack.amazonaws.com/PR-$PR_ID/$BUILD_ID/index.html
- [ ] To the my best knowledge, examples are either not affected by this change, or have been fixed to be compatible with this change

### Changes ###
- [ ] Feature1, tests, (and when applicable, API doc)
- [ ] Feature2, tests, (and when applicable, API doc)

## Comments ##
- If this change is a backward incompatible change, why must this change be made.
- Interesting edge cases to note here

### CODE REVIEW

File: src/ngraph/ngraph_emitter.cc [hunk 1]
```diff
@@ -204,6 +204,21 @@ void Emitter::CreateUnaryOps() {
   // ngraph_op_funcs_["log_softmax"] = [this](const NodePtr& node){
   //   return ;
   // };
+  ngraph_op_funcs_["SoftmaxActivation"] = [this](const NodePtr& node) {
+    ngraph::AxisSet axes;
+
+    auto in_shape = op_map_[node->inputs_[0]]->get_shape();
+    auto mode = get_default(node, "mode", std::string("instance"));
+
+    if (mode == std::string("channel")) {
```
Comment (cconvey) [inline src/ngraph/ngraph_emitter.cc]: Why the explicit use of `std::string`?

File: tests/python/ngraph/test_corner_cases.py [hunk 2]
```diff
@@ -51,6 +51,221 @@ def test_broadcast_mix_logic_op():
...
+        [0.23905596, 0.2512115 , 0.18117926, 0.16931345, 0.15923984]],
+
+       [[0.22951414, 0.16428624, 0.23097359, 0.17542477, 0.19980127],
+        [0.24256574, 0.24258773, 0.1683444 , 0.1855961 , 0.16090597],
+        [0.19252302, 0.18172522, 0.29073915, 0.20903045, 0.12598222],
+        [0.28281614, 0.19658898, 0.12981647, 0.12611076, 0.26466766],
+        [0.12840787, 0.21151797, 0.17961076, 0.23257242, 0.24789092]]],
+      dtype=np.float32)
+    assert np.allclose(forward, true_forward)
+    # Don't run backwards because mxnet is wrong for this shape and option
```
Comment (cconvey) [inline tests/python/ngraph/test_corner_cases.py]: NBD, but it would be nice if you could give a little more detail here (bug-tracker URL, or summary, etc.) in case people want to follow-up on this.  If I understand the comment correctly, this is a hole in our unit tests that ideally will be filled in at a later date.

Comment (mbrookhart) [reply tests/python/ngraph/test_corner_cases.py]: It's a hole in the unit tests of public MXNet for a deprecated op, they don't plan to support it moving forward, we're only doing this to make certain legacy models work correctly in inference. I'll expand the comment.

File: src/ngraph/ngraph_emitter.cc (hunk 1 above)
Comment (mbrookhart) [reply src/ngraph/ngraph_emitter.cc]: I was getting odd compilation errors related to booleans, so I just made it explicit.

**Comment by mbrookhart** [comment]
TESTNOW
